src/Compiler/Checking/Expressions/CheckComputationExpressions.fs

```diff
@@ -1900,105 +1900,162 @@ let rec TryTranslateComputationExpression
-            if isNil andBangs then
-                error (Error(FSComp.SR.tcInvalidUseBangBinding (), mBind))
-            else
-                let m =
-                    match andBangs with
-                    | [] -> comp.Range
-                    | h :: _ -> h.Trivia.AndBangKeyword
+                    error (Error(FSComp.SR.tcInvalidUseBangBindingNoAndBangs (), m))
+                else
+                    if ceenv.isQuery then
+                        error (Error(FSComp.SR.tcBindMayNotBeUsedInQueries (), mBind))
+
+                    if
+                        isNil (
+                            TryFindIntrinsicOrExtensionMethInfo
+                                ResultCollectionSettings.AtMostOneResult
+                                cenv
+                                ceenv.env
+                                mBind
+                                ceenv.ad
+                                "Using"
+                                ceenv.builderTy
+                        )
+                    then
+                        error (Error(FSComp.SR.tcRequireBuilderMethod "Using", mBind))
+
+                    if
+                        isNil (
+                            TryFindIntrinsicOrExtensionMethInfo
+                                ResultCollectionSettings.AtMostOneResult
+                                cenv
+                                ceenv.env
+                                mBind
+                                ceenv.ad
+                                "Bind"
+                                ceenv.builderTy
+                        )
+                    then
+                        error (Error(FSComp.SR.tcRequireBuilderMethod "Bind", mBind))
+
+                    let bindExpr =
+                        let consumeExpr =
+                            SynExpr.MatchLambda(
+                                false,
+                                mBind,
+                                [
+                                    SynMatchClause(
+                                        pat,
+                                        None,
+                                        TranslateComputationExpressionNoQueryOps ceenv innerComp,
+                                        innerComp.Range,
+                                        DebugPointAtTarget.Yes,
+                                        SynMatchClauseTrivia.Zero
+                                    )
+                                ],
+                                DebugPointAtBinding.NoneAtInvisible,
+                                mBind
+                            )
+
+                        let rhsExpr =
+                            mkSourceExprConditional isFromSource rhsExpr ceenv.sourceMethInfo ceenv.builderValName
 
-                error (Error(FSComp.SR.tcInvalidUseBangBindingNoAndBangs (), m))
+                        mkSynCall "Bind" mBind [ rhsExpr; consumeExpr ] ceenv.builderValName
```

Comment:
I was expecting a call to "Using" somewhere, but cannot find it.
Is it implicit by recursive calling into a different handler?

Reply:
The way I understood the logic was that `_` (wild) will need to be `Bind` in the same way that  `ident`(any ident) `__` (double underscore). But Im not very familiar with the CE Translation logic.

Tested this to `"Using"` and it does not work.

Reply:
If it only does "Bind" and not an "Using", it will lack the CE's implementation of disposal (which can be anything the CE wants to do to cleanup).